system-addon/lib/MessageCenterRouter.jsm

```diff
@@ -132,18 +138,22 @@ class _MessageCenterRouter {
         await this.sendNextMessage(target);
         break;
       case "BLOCK_MESSAGE_BY_ID":
-        await this.setState(state => {
-          const newState = Object.assign({}, state.blockList);
-          newState[action.data.id] = true;
-          return {blockList: newState};
+        await this.setState(async state => {
+          const blockList = state.blockList.slice();
```

Comment:
personally I think `[...state.blockList]` more clearly indicates you're making a copy